src/Parser/EnhancementShaman/CombatLogParser.js

```diff
@@ -0,0 +1,336 @@
+    const stormBringer = getAbility(SPELLS.STORMBRINGER.id);
+
+    const fightDuration = this.fightDuration;
+
+	  //uptimes
+	  const flametongueUptime = this.selectedCombatant.getBuffUptime(SPELLS.FLAMETONGUE_BUFF.id) / this.fightDuration;
+	  const frostbrandUptime = this.selectedCombatant.getBuffUptime(SPELLS.FROSTBRAND.id) / this.fightDuration;
+    const landslideUptime = this.selectedCombatant.getBuffUptime(SPELLS.LANDSLIDE_BUFF.id) / this.fightDuration;
+    const furyofairUptime = this.selectedCombatant.getBuffUptime(SPELLS.FURY_OF_AIR.id) / this.fightDuration;
+
+    const nonDpsTimePercentage = this.modules.alwaysBeCasting.totalDamagingTimeWasted / fightDuration;
+    const deadTimePercentage = this.modules.alwaysBeCasting.totalTimeWasted / fightDuration;
+
+    const masteryPercent = this.selectedCombatant.masteryPercentage;
+    const hastePercent = this.selectedCombatant.hastePercentage;
+
+    //Legendaries
+    // const hasUncertainReminder = this.selectedCombatant.hasHead(ITEMS.UNCERTAIN_REMINDER.id);
+    // const hasEmalons = this.selectedCombatant.hasChest(ITEMS.EMALONS_CHARGED_CORE.id);
+    const hasAkainus = this.selectedCombatant.hasWrists(ITEMS.AKAINUS_ABSOLUTE_JUSTICE.id);
+    // const hasSmolderingHeart = this.selectedCombatant.hasHands(ITEMS.SMOLDERING_HEART.id);
+    // const hasStorm = this.selectedCombatant.hasWaist(ITEMS.STORM_TEMPESTS.id);
+    // const hasRoots = this.selectedCombatant.hasLegs(ITEMS.ROOTS_OF_SHALADRASSIL.id);
+    // const hasSpiritual = this.selectedCombatant.hasFeet(ITEMS.SPIRITUAL_JOURNEY.id);
+    // const hasSoul = this.selectedCombatant.hasFinger(ITEMS.SOUL_OF_THE_FARSEER.id);
+    const hasEOTN = this.selectedCombatant.hasFinger(ITEMS.EYE_OF_THE_TWISTING_NETHER.id);
+    // const hasSephuz = this.selectedCombatant.hasFinger(ITEMS.SEPHUZS_SECRET.id);
+
+  this.selectedCombatant._combatantInfo.gear.forEach(function(value) {
+    const equippedItem = getItemInfo(value.id);
+
+    if(equippedItem !== undefined && equippedItem.quality === 5) {
+      results.items.push({
+        item: equippedItem,
+        result: (
+          <dfn data-tip="">
+            Equipped Legendary
+          </dfn>
+        ),
+      });
+    }
+  });
+
+  //add trinkets to item list
+  function getItemInfo(id) {
+    let value;
+    Object.keys(ITEMS).some(function(k) {
+      if(ITEMS[k].id === id) {
+        value = ITEMS[k];
+      }
+    });
+
+    return value;
+  }
+
+    if (nonDpsTimePercentage > 0.3) {
+      results.addIssue({
+        issue: `[NYI] Your non DPS time can be improved. Try to cast damaging spells more regularly (${Math.round(nonDpsTimePercentage * 100)}% non DPS time).`,
+        icon: 'petbattle_health-down',
+        importance: getIssueImportance(nonDpsTimePercentage, 0.4, 0.45, true),
+      });
+    }
+    if (deadTimePercentage > 0.2) {
+      results.addIssue({
+        issue: `Your dead GCD time can be improved. (${Math.round(deadTimePercentage * 100)}% dead GCD time).`,
```

Comment:
Suggestions are supposed to also provide a suggestion to improve when possible. This could be done by changing this sentence slightly to something like `Your dead GCD time can be improved. Try to Always Be Casting (ABC); cast instant spells during movement phases and focus on having no delays between spell casts.`.